### PR TITLE
Exclude explicitly implemented interface methods from proxy

### PR DESCRIPTION
**Pull request by alrz**
Fix https://github.com/dotnet/orleans/issues/8991
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/orleans/pull/8992)

### CODE REVIEW

File: src/Orleans.CodeGenerator/Model/ProxyInterfaceDescription.cs [hunk 1]
```diff
@@ -81,7 +81,12 @@ private List<ProxyMethodDescription> GetMethods(INamedTypeSymbol symbol)
             {
                 foreach (var method in iface.GetDeclaredInstanceMembers<IMethodSymbol>())
                 {
-                    if (methods.TryGetValue(method, out _))
+                    if (method.MethodKind == MethodKind.ExplicitInterfaceImplementation)
```
Comment (alrz) [inline src/Orleans.CodeGenerator/Model/ProxyInterfaceDescription.cs:84]: I think this should also check for a body, explicit impls can be used for reabstraction in interface type hierarchies.

**Comment by alrz** [comment]
What sort of tests do we need here? I'd appreciate some pointers on existing unit tests so I can follow from there. Also not sure about namings and the location these interfaces should be declared in.

@ReubenBond for review, thanks.

**Comment by ReubenBond** [comment]
I fixed the merge conflict, but it essentially removed your change. I think this is something we don't want to do, since we want people to be able to override those explicit interface implementations on their grain classes. If we skipped them in the proxy, then that wouldn't be possible: the base implementation would always be called (and it would be called at the caller's side, which is not right).

File: src/Orleans.CodeGenerator/Model/ProxyInterfaceDescription.cs (hunk 1 above)
Comment (ReubenBond) [reply src/Orleans.CodeGenerator/Model/ProxyInterfaceDescription.cs:82]: I am not sure it's necessary: the generated proxy should implement every method explicitly (I added a comment below this line). Any overrides should take effect on the callee side naturally (correct me if you find that I'm wrong here)

**Pull request merged**